### ISSUE TITLE
Default to parsing command-line arguments for yargs factory too

### ISSUE THREAD

**Issue by shadowspawn** [yargs/yargs#2399]
The old singleton pattern defaulted to the most common case of parsing the command-line arguments:

```
const yargs = require('yargs');
const argv = yargs.parse();
```

The new factory pattern is currently a bit clunky by comparison.
```
const yargs = require('yargs/yargs');
const {hideBin} = require('yargs/helpers');
const argv = yargs(hideBin(process.argv)).parse();
```

If we default to the command-line arguments we get back to close to the singleton ergonomics. Single import. Simple creation.
```
const yargs = require('yargs/yargs');
const argv = yargs().parse();
```

It seems reasonably easy and potentially non-breaking. Am I missing something?

(I wonder if retaining the default behaviour got left out or was more difficult early on with the big refactors to move to the factory pattern and multi-platform node/deno/browser.)

**Comment by ljharb** [comment]
I agree; especially since it's not an error to pass _nothing_, the lack of defaulting creates a very confusing footgun for those migrating to v18.
